### PR TITLE
Added new method for discarding cards to handle batch triggers (ready for review)

### PR DESCRIPTION
**Pull request by theelk801**
In order to make Rielle, the Everwise (and likely other cards in the future) work we need to refactor how batch discarding works, currently working through anything that discards multiple cards at once using the discard a single card method.

### CODE REVIEW

**Comment by JayDi85** [comment]
I need time to complete new review (there are some missings and improves). Will finish it later (today or tomorrow).

File: Mage.Sets/src/mage/cards/s/SkullRend.java [hunk 1]
```diff
@@ -38,44 +34,31 @@ public SkullRend copy() {
...
-                        }
-                    }
+            for (UUID playerId : game.getOpponents(source.getControllerId())) {
+                Player opponent = game.getPlayer(playerId);
+                if (opponent == null) {
+                    continue;
                 }
-                return true;
+                opponent.damage(2, source.getSourceId(), game);
+                opponent.discard(2, true, source, game);
```
Comment (JayDi85) [inline Mage.Sets/src/mage/cards/s/SkullRend.java:54]: Maybe card's bug: I think there should be two cycles (damage all first, then discard all -- as card's rules).

File: Mage.Sets/src/mage/cards/w/WitsEnd.java [hunk 2]
```diff
@@ -59,11 +55,7 @@ public WitsEndEffect copy() {
     public boolean apply(Game game, Ability source) {
         Player player = game.getPlayer(source.getFirstTarget());
         if (player != null) {
-            Set<Card> cards = player.getHand().getCards(game);
-            for (Card card : cards) {
-                player.discard(card, source, game);
-            }
-            return true;
+            player.discard(player.getHand(), source, game);
```
Comment (JayDi85) [inline Mage.Sets/src/mage/cards/w/WitsEnd.java:58]: Missing return true

File: Mage.Sets/src/mage/cards/w/WrenchMind.java [hunk 3]
```diff
@@ -59,18 +57,19 @@ public WrenchMindEffect copy() {
...
-                if (!card.isArtifact() && !targetPlayer.getHand().isEmpty()) {
-                    targetPlayer.discard(1, false, source, game);
-                }
-                return true;
-            }            
+        if (targetPlayer == null || targetPlayer.getHand().isEmpty()) {
+            return false;
+        }
+        if (targetPlayer.getHand().count(StaticFilters.FILTER_CARD_ARTIFACT, game) < 1
+                || !targetPlayer.chooseUse(outcome, "Discard an artifact card?", source, game)) {
```
Comment (JayDi85) [inline Mage.Sets/src/mage/cards/w/WrenchMind.java:64]: Must be Outcome.Benefit (AI must try to discard 1 card instead 2)

File: Mage.Sets/src/mage/cards/n/NantukoCultivator.java [hunk 4]
```diff
@@ -1,42 +1,41 @@
...
-        super(ownerId,setInfo,new CardType[]{CardType.CREATURE},"{3}{G}");
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{3}{G}");
         this.subtype.add(SubType.INSECT);
         this.subtype.add(SubType.DRUID);
         this.power = new MageInt(2);
         this.toughness = new MageInt(2);
 
         // When Nantuko Cultivator enters the battlefield, you may discard any number of land cards. Put that many +1/+1 counters on Nantuko Cultivator and draw that many cards.
-        this.addAbility(new EntersBattlefieldTriggeredAbility(new NantukoCultivatorEffect(), true));
+        this.addAbility(new EntersBattlefieldTriggeredAbility(new NantukoCultivatorEffect()));
```
[the remark this answers is not in the document]
Comment (theelk801) [reply Mage.Sets/src/mage/cards/n/NantukoCultivator.java:35]: not necessary, the text already has "you may" and the the targeting lets you choose zero cards

**Pull request merged**